tensorstore/kvstore/s3/s3_uri_utils.h

```diff
@@ -0,0 +1,61 @@
+// Copyright 2023 The TensorStore Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TENSORSTORE_KVSTORE_S3_URI_UTILS_H
+#define TENSORSTORE_KVSTORE_S3_URI_UTILS_H
+
+#include <string>
+#include <string_view>
+
+#include "tensorstore/internal/uri_utils.h"
+
+using ::tensorstore::internal::AsciiSet;
```

Comment:
A style note: Google internal style forbids the use of anonymous namespaces and using declarations in headers.

Anonymous namespaces in headers can cause duplicate symbols in codegen.

In this case they're not really useful/necessary anyway, as it's fine to put these constants into the internal_http_s3 namespace.

Reply:
Will also fix `using` in the s3 headers polluting namespaces in conjunction with a general `internal_kvstore_s3` namespace strategy.

Reply:
BTW Thanks for reminding me of this (no `using directives` in headers). I'm running into all the sharp edges after a bit of a hiatus.

Reply:
All s3 functionality has been moved into the `internal_kvstore_s3` namespace and `using` in headers has been removed.